### PR TITLE
Ssl rewrite

### PR DESCRIPTION
**Pull request by mcalmer**
# Description

Introduce a new tool to setup SSL certificates on Server an Proxy.

# Target branches

Which documentation version does this PR apply to?

- [x] Master (Default)
- [ ] Manager-4.2
- [ ] Manager-4.1
- [ ] Manager-4.0

# Links

Code PR https://github.com/uyuni-project/uyuni/pull/4454

### CODE REVIEW

File: modules/administration/pages/ssl-certs-imported.adoc [hunk 1]
```diff
@@ -1,27 +1,28 @@
...
 
-This section covers how to use an imported SSL certificate with a new {productname} installation, and how to replace existing self-signed certificates with imported certificates.
+This section covers how to configure SSL certificate for new {productname} installation, and how to replace existing certificates.
 
 Before you begin, ensure you have:
 
 * A certificate authority (CA) SSL public certificate.
-  If you are chaining CAs, it must include the root CA, and all intermediate CAs.
-* An SSL server key
+  If you are using a CA chain, all intermediate CAs must be available as well.
```
Comment (keichwa) [inline modules/administration/pages/ssl-certs-imported.adoc:12]:
```suggestion
  If you are using a CA chain, all intermediate CAs must also be available.
```

File: modules/administration/pages/ssl-certs-imported.adoc [hunk 2]
```diff
@@ -1,27 +1,28 @@
...
+The host name of the SSL server certificate must match the fully qualified host name of the machine you deploy them on.
 You can set the host names in the ``X509v3 Subject Alternative Name`` section of the certificate.
 You can also list multiple host names if your environment requires it.
 
 Third-party authorities commonly use intermediate CAs to sign requested server certificates. 
-In this case, all CAs in the chain are required to be available in a single file to deploy the CA to the clients.
-
+In this case, all CAs in the chain are required to be available.
+If there is not extra parameter or option available to specify intermediate CAs, take care that all CAs - Root CA and intermediate CAs -
+are available in one file.
```
Comment (keichwa) [inline modules/administration/pages/ssl-certs-imported.adoc:25]:
```suggestion
If there is no extra parameter or option available to specify intermediate CAs, take care that all CAs (Root CA and intermediate CAs) are stored in one file.
```

File: modules/administration/pages/ssl-certs-imported.adoc [hunk 3]
```diff
@@ -86,61 +87,63 @@ Certificates signed with different CAs do not match.
...
+== Replace Certificates
 
+You can replace active certificates on your {productname} installation with a new certificate.
+To replace the certificates, you can replace the installed CA certificate with the new CA, and then update the database.
 
 
 .Procedure: Replacing Existing Certificates
 
-. On the {productname} Server, at the command prompt, move the old certificate directory to a backup location:
+. On the {productname} Server, at the command prompt, and call the command `mgr-ssl-cert-setup` and provide the certificates as parameter:
```
Comment (keichwa) [inline modules/administration/pages/ssl-certs-imported.adoc:98]:
```suggestion
. On the {productname} Server, at the command prompt, call the command `mgr-ssl-cert-setup` and provide the certificates as parameters:
```

File: modules/administration/pages/ssl-certs-imported.adoc [hunk 4]
```diff
@@ -86,61 +87,63 @@ Certificates signed with different CAs do not match.
...
 
-. Generate a new server certificate RPM:
-+
-----
-rhn-ssl-tool --gen-server --rpm-only --dir="/root/ssl-build" --from-server-key=<Server_Key_File> --from-server-cert=<Server_Cert_File>
-----
+Intermediate CAs can either be available in the file which is specified as 'root-ca-file' or specified by extra options `--intermediate-ca-file`.
+This option can be used multiple times.
+
+This command performs a number of tests on the provides files to test if they are valid and can be used for the requested use case.
```
Comment (keichwa) [inline modules/administration/pages/ssl-certs-imported.adoc:108]:
```suggestion
Intermediate CAs can either be available in the file which is specified with `--root-ca-file` or specified as extra options with `--intermediate-ca-file`.
The `--intermediate-ca-file` option can be specified multiple times.
This command performs a number of tests on the provided files to test if they are valid and can be used for the requested use case.
```

File: modules/administration/pages/ssl-certs-imported.adoc [hunk 5]
```diff
@@ -86,61 +87,63 @@ Certificates signed with different CAs do not match.
...
-When you have created the new [path]``ssl-build`` directory, you can create combined certificate RPMs and deploy them on the clients.
-For the procedures to do this, see xref:administration:ssl-certs-selfsigned.adoc[].
 
+=== Extra handling of traditional Clients
+
+Traditional Clients are deprecated and should be replaced with Salt Client.
+
+If the CA needs to be replaced when there are still traditional managed clients connected to {productname}
+some extra steps are required.
+
```
Comment (keichwa) [inline modules/administration/pages/ssl-certs-imported.adoc:141]:
```suggestion
If the CA needs to be replaced when there are still traditional managed clients connected to {productname} some extra steps are required.

```

File: modules/administration/pages/ssl-certs-imported.adoc [hunk 6]
```diff
@@ -86,61 +87,63 @@ Certificates signed with different CAs do not match.
...
+Important is, that the clients do not get disconnected when the new CA is activated on the {productname} Server and Proxies.
+Deploy the 'old' and the 'new' Root CA certificates to the affected clients and trust them.
+Use a configuration channel to deploy the certificte files to the clients and the remote command feature to regenerate
+the trust store.
```
Comment (keichwa) [inline modules/administration/pages/ssl-certs-imported.adoc:145]:
```suggestion
Use a configuration channel to deploy the certificte files to the clients and the remote command feature to regenerate the trust store.
```

File: modules/administration/pages/ssl-certs-imported.adoc [hunk 7]
```diff
@@ -86,61 +87,63 @@ Certificates signed with different CAs do not match.
...
+
+After the new certificates are activated on the {productname} Server and Proxies, test if the connections are working
+and actions can still be scheduled on the clients. If this is the case, the 'old' Root CA can be removed from clients.
```
Comment (keichwa) [inline modules/administration/pages/ssl-certs-imported.adoc:148]:
```suggestion
After the new certificates are activated on the {productname} Server and Proxies, test if the connections are working and actions can still be scheduled on the clients.
If this is the case, the 'old' Root CA can be removed from clients.
```

File: modules/administration/pages/ssl-certs.adoc [hunk 8]
```diff
@@ -36,20 +36,28 @@ If you are using intermediate CAs in a chained certificate, the root CA is insta
...
+//It should not contain the server certificate.
+//The server certificate must be defined at the ``SERVER_CERT`` environment variable.
 
-By default, {productname} uses a self-signed certificate.
+By default, {productname} uses a self-signed certificate without intermediate CAs.
 For additional security, you can arrange a third party CA to sign your certificates.
 Third party CAs perform checks to ensure that the information contained in the certificate is correct.
 They usually charge an annual fee for this service.
 Using a third party CA makes certificates harder to spoof, and provides additional protection for your installation.
 If you have certificates signed by a third party CA, you can import them to your {productname} installation.
```
Comment (keichwa) [inline modules/administration/pages/ssl-certs.adoc:54]: // FIXME: I must check the following paras when this PR is ready for review. -ke-

**Review by keichwa — APPROVED** [review]

**Review by keichwa — APPROVED** [review]

**Review by jcayouette — APPROVED** [review]

**Comment by keichwa** [comment]
Please check the .changelog entry I just added.  The rest LGTM!

**Pull request merged**